### ISSUE TITLE
Routing error when not launching app directly into the blog

### ISSUE THREAD

**Issue by jamesosb** [Bodacious/blogit#10]
Hello again! 

I'm facing an issue when I set the root to outside the blog. If I start the app on a page other than /blog then I get the following routing error:

```
undefined method `blogit_authenticate' for Blogit::PostsController:Class
```

refreshing the page then delivers the following NoMethodError:

```
undefined method `current_page' for nil:NilClass
```

I took a look at the recent bug fixes but the one resembling a solution to this error was a tweak to the Posts Controller. I implemented this fix but am still getting these errors. 

Any thoughts? 

Thanks,

James

**Comment by philou** [comment]
Hi,

I don't know exactly what this could be. I am not facing this issue at all, although I have set root to outside the blog. I fixed something that looked like this for version 0.5.0 (commit 7a251f4 actually). 

Here are just a few ideas you might dig into :
- the issue I fixed was about autoloading ::Blogit::ApplicationController, maybe it could be related to this ?
- In my working app, I have declared the root route before mounting the blog engine, did you try to invert root and blog routes ?

Philippe
